File: src/app/Service/dto.ts
export class Dto{
    public x!: number;
    public y!: number;
    public type!: string;
    public color!: string;
    public id!: string;
    public text!: string;
    public machineToQueue: string[] = []
    public queueToMachine: string[] = []
    public rootGraph: Dto[] = [];
    public productsNumberInStock: number = 10;
    public numberOfProducts!: number;
}

File: src/main/java/com/producer_consumer/DTOs/Dto.java
package com.producer_consumer.DTOs;

import java.util.List;

public class Dto {
    public int x;
    public int y;
    public int numberOfProducts;
    public int productsNumberInStock;
    public String type;
    public String color;
    public String id;
    public String text;
    public List<String> machineToQueue;// all ids of machine that connected as input to that queue
    public List<String> queueToMachine;// all ids of machine that connected as output to that queue
    public List<Dto> rootGraph;
}